### PR TITLE
Task #79217 Edit profile

### PR DESCRIPTION
**Pull request by IamRedsan**
## Related Tickets
- [Task #79217](https://edu-redmine.sun-asterisk.vn/issues/79217)

## HOW
- Update avatar
- Update thông tin user
- Update mật khẩu

## Evidence (Screenshot or Video)

https://github.com/user-attachments/assets/384b8617-560a-4594-b5ac-cde408bce287

https://github.com/user-attachments/assets/d7234daf-f906-4884-84cf-f3f7a59d4ae3

https://github.com/user-attachments/assets/17b85774-075e-43d0-8531-2750320419a5

### CODE REVIEW

**Comment by IamRedsan** [comment]
ready

File: app/forms.py [hunk 1]
```diff
@@ -69,3 +69,63 @@ def save(self, commit=True):
...
+
+class UserProfileForm(forms.ModelForm):
+    class Meta:
+        model = CustomUser
+        fields = ['first_name', 'last_name',
+                  'default_address', 'default_phone_number']
+
+    def clean_phone_num(self):
+        phone_num = self.cleaned_data.get('default_phone_number')
+        phone_pattern = re.compile(r'^0\d{9}$')
```
Comment (huongnt-2545) [inline app/forms.py]: `r'^0\d{9}$'` -> khai báo trong constants

File: app/views.py [hunk 2]
```diff
@@ -536,3 +538,75 @@ def profile_view(request, pk):
...
+
+@login_required
+def update_profile(request):
+    user = request.user
+    if request.method == 'POST':
+        form = UserProfileForm(request.POST, instance=user)
+        if form.is_valid():
+            form.save()
+            messages.success(
+                request, 'Your profile has been updated successfully!')
```
Comment (huongnt-2545) [inline app/views.py]: translate message

File: app/views.py [hunk 3]
```diff
@@ -536,3 +538,75 @@ def profile_view(request, pk):
...
+    user = request.user
+    if request.method == 'POST':
+        form = AvatarUploadForm(request.POST, request.FILES, instance=user)
+        if 'image' in request.FILES and request.FILES['image'].size > 0:
+            uploaded_file = request.FILES['image']
+            upload_result = cloudinary.uploader.upload(uploaded_file)
+            user.avatar = upload_result['secure_url']
+            user.save()
+            messages.success(
+                request, 'Your avatar has been updated successfully!')
```
Comment (huongnt-2545) [inline app/views.py]: translate message

**Comment by IamRedsan** [comment]
ready

**Review by huongnt-2545 — COMMENTED** [review]
conflicted

**Comment by IamRedsan** [comment]
ready

**Comment by IamRedsan** [comment]
ready

**Comment by IamRedsan** [comment]
ready

**Comment by huongnt-2545** [comment]
conflicted

**Comment by IamRedsan** [comment]
ready

**Comment by IamRedsan** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]

**Pull request merged**